docs/src/service-workers-guide.md

```diff
@@ -0,0 +1,159 @@
+---
+id: service-workers-guide
+title: "Service Workers Guide"
+---
+
+## Disabling Service Workers
+
+[Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) provide a browser-native way to handle requests made with the native [Fetch API (`fetch`)](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
+
+If you're using Playwright's [Network-related features](./network.md) like the routing APIs ([`method: BrowserContext.route`], [`method: BrowserContext.routeFromHAR`], [`method: Page.route`], [`method: Page.routeFromHAR`]), or rely heavily on Network Events (e.g. [`event: Page.request`], [`event: BrowserContext.request`], etc.), we recommend disabling Service Workers by setting [`option: Browser.newContext.serviceWorkers`] to `'block'`.
```

Comment:
I'm not sure I agree with this recommendation. The qualification helps, but there is a point to be made this potentially changes the behavior of the app under test s.t. it no longer reflects what a user would experience.

Reply:
I think the paragraph about FF/WK is the most important one. If you use networking APIs and test in these browsers, we strongly recommend to disable SW.